index.js

```diff
@@ -59,21 +78,29 @@ export default class OneSignal {
         // Listen to events of notification received, opened, device registered and IDSAvailable.
 
         invariant(
-            type === 'received' || type === 'opened' || type === 'ids' || type == 'emailSubscription',
-            'OneSignal only supports `received`, `opened`, and `ids` events'
+            type === NOTIFICATION_RECEIVED_EVENT ||
+            type === NOTIFICATION_OPENED_EVENT ||
+            type === IDS_AVAILABLE_EVENT ||
+            type === EMAIL_SUBSCRIPTION_EVENT ||
+            type === IN_APP_MESSAGE_CLICKED_EVENT,
+            'OneSignal only supports `received`, `opened`, `ids`, `emailSubscription`, and `inAppMessageClicked` events'
         );
 
         _eventTypeHandler.set(type, handler);
 
-        if (type == 'opened') {
-            RNOneSignal.didSetNotificationOpenedHandler();
+        if (type === NOTIFICATION_OPENED_EVENT) {
+            RNOneSignal.initNotificationOpenedHandlerParams();
```

Comment:
Any name changes to `RNOneSignal` function need their native parts updated. I see Android was changed but we need to update iOS name too.

Reply:
This was me and I noted myself to do this, but thank you for catching this